pkg/hvs/controllers/flavortemplate_controller_test.go

```diff
@@ -0,0 +1,434 @@
+				req.Header.Set("Accept", consts.HTTPMediaTypeJson)
+				req.Header.Set("Content-Type", consts.HTTPMediaTypeJson)
+				Expect(err).NotTo(HaveOccurred())
+				w = httptest.NewRecorder()
+				router.ServeHTTP(w, req)
+				Expect(w.Code).To(Equal(http.StatusOK))
+			})
+		})
+
+		Context("Retrieve data from not available FlavorTemplate ID", func() {
+			It("Should not retrieve Flavortemplate data and get HTTP Status: 404", func() {
+				router.Handle("/flavor-template/{id}", hvsRoutes.ErrorHandler(hvsRoutes.JsonResponseHandler(flavorTemplateController.Retrieve))).Methods("GET")
+				req, err := http.NewRequest("GET", "/flavor-template/73755fda-c910-46be-821f-e8ddeab189e9", nil)
+				req.Header.Set("Accept", consts.HTTPMediaTypeJson)
+				req.Header.Set("Content-Type", consts.HTTPMediaTypeJson)
+				Expect(err).NotTo(HaveOccurred())
+				w = httptest.NewRecorder()
+				router.ServeHTTP(w, req)
+				Expect(w.Code).To(Equal(http.StatusNotFound))
+			})
+		})
+
+		Context("Retrieve data from invalid FlavorTemplate ID", func() {
+			It("Should not retrieve Flavortemplate data and get HTTP Status: 404", func() {
+				router.Handle("/flavor-template/{id}", hvsRoutes.ErrorHandler(hvsRoutes.JsonResponseHandler(flavorTemplateController.Retrieve))).Methods("GET")
+				req, err := http.NewRequest("GET", "/flavor-template/2f5cf0ec-0000-0000-0000-000000000000", nil)
+				req.Header.Set("Accept", consts.HTTPMediaTypeJson)
+				req.Header.Set("Content-Type", consts.HTTPMediaTypeJson)
+				Expect(err).NotTo(HaveOccurred())
+				w = httptest.NewRecorder()
+				router.ServeHTTP(w, req)
+				Expect(w.Code).To(Equal(http.StatusNotFound))
+			})
+		})
+
+	})
+
+	Describe("Search And Delete Flavor Templates", func() {
+		Context("When no filter arguments are passed", func() {
+			It("All Flavor template records are returned", func() {
+				router.Handle("/flavor-template/", hvsRoutes.ErrorHandler(hvsRoutes.JsonResponseHandler(flavorTemplateController.Search))).Methods("GET")
+				req, err := http.NewRequest("GET", "/flavor-template/", nil)
+				Expect(err).NotTo(HaveOccurred())
+				req.Header.Set("Accept", consts.HTTPMediaTypeJson)
+				w = httptest.NewRecorder()
+				router.ServeHTTP(w, req)
+				Expect(w.Code).To(Equal(http.StatusOK))
+
+				var ft *[]hvs.FlavorTemplate
+				err = json.Unmarshal(w.Body.Bytes(), &ft)
+				Expect(err).ToNot(HaveOccurred())
+			})
+		})
+		Context("Delete a template", func() {
+			It("The templates should be deleted", func() {
+				router.Handle("/flavor-template/{id}", hvsRoutes.ErrorHandler(hvsRoutes.JsonResponseHandler(flavorTemplateController.Delete))).Methods("DELETE")
+				req, err := http.NewRequest("DELETE", "/flavor-template/426912bd-39b0-4daa-ad21-0c6933230b51", nil)
+				Expect(err).NotTo(HaveOccurred())
+				req.Header.Set("Accept", consts.HTTPMediaTypeJson)
+				w = httptest.NewRecorder()
+				router.ServeHTTP(w, req)
+				Expect(w.Code).To(Equal(http.StatusNotFound))
+			})
+		})
+		Context("Delete a template", func() {
```

Comment:
Search And Delete Flavor Templates -> Delete a template ->The templates should be deleted

Does not sound right